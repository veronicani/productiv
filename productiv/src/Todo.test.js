import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

const TEST_TODO = {
  id: 1,
  title: 'Test',
  description: 'Test Description',
  priority: 1};

describe("productiv app", function () {
  it("renders without crashing", function () {
    render(<Todo todo={TEST_TODO}/>);
  });

  it("contains expected title", function () {
    const { container, debug } = render(<Todo todo={TEST_TODO}/>);
    const todoDiv = container.querySelector(".Todo");
    expect(todoDiv).toContainHTML("<b>Test</b>");
  });

  it("contains expected description", function () {
    const result = render(<Todo todo={TEST_TODO}/>);
    expect(result.queryByText("Test Description")).toBeInTheDocument();
  });

  it("contains expected priority", function () {
    const result = render(<Todo todo={TEST_TODO}/>);
    expect(result.queryByText("(priority: 1)")).toBeInTheDocument();
  });

  //TODO: SNAPSHOT
});