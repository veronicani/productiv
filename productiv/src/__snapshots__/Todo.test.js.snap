// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        Test
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        Test Description
      </small>
    </div>
  </div>
</div>
`;

exports[`productiv app matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        Test
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        Test Description
      </small>
    </div>
  </div>
</div>
`;
